pkg/autoscaler/stats_scraper.go

```diff
@@ -127,11 +133,50 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 		return nil, nil
 	}
 
-	stat, err := s.sClient.Scrape(s.url)
-	if err != nil {
-		return nil, err
+	sampleSize := populationMeanSampleSize(readyPodsCount)
+
+	var (
```

Comment:
What if:

-  we had `type stat struct {... stats go here... }`
- we have channel of `*stat`
- once each of the scrapers scrapes it pushes stuff into the channel
- outside loop reads the channel (can be either parallel or synchronous and channel buffered).

More go style, one mutex less

Reply:
I agree on that. Alternatively you can create an array like this `[sampleSize]*Stat` and write each stat at `i` and process afterwards.

Reply:
Changed to using channels.